File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    p1=[1, 2, 3, 4, 5]
    p2=[2, 1, 2, 3, 2, 4, 2, 5]
    p3=[3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    ps1=0
    ps2=0
    ps3=0
    n=len(answers)
    max_r=0
    for i in range(n):
        if answers[i]==p1[i%len(p1)]:
            ps1+=1
        if answers[i]==p2[i%len(p2)]:
            ps2+=1     
        if answers[i]==p3[i%len(p3)]:
            ps3+=1
    max_r=max(ps1,ps2,ps3)
    if max_r==0:
        return answer
    if ps1==max_r:
        answer.append(1)
    if ps2==max_r:
        answer.append(2)   
    if ps3==max_r:
        answer.append(3)
    answer.sort()
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    
    vector<int> p1 = {1, 2, 3, 4, 5};
    vector<int> p2 = {2, 1, 2, 3, 2, 4, 2, 5};
    vector<int> p3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    
    int ps1 = 0;
    int ps2 = 0;
    int ps3 = 0;
    
    int n = answers.size();
    for(int i=0;i<n;i++){
        if (answers[i]==p1[i%p1.size()]){
            ps1++;
        }
        
        if (answers[i]==p2[i%p2.size()]){
            ps2++;
        }
        if (answers[i]==p3[i%p3.size()]){
            ps3++;
        }
    }
    int max_v = 0;
    max_v=max({ps1,ps2,ps3});
    if (max_v==ps1){
        answer.push_back(1);
    }
    if (max_v==ps2){
        answer.push_back(2);
    }
    if (max_v==ps3){
        answer.push_back(3);
    }
    
        
        return answer;
}